File: Cpp/TirePressureMonitoring/alarm.cpp
#include "alarm.hpp"
#include "sensor.hpp"

Alarm::Alarm()
{
    sensor = Sensor();
    lowPressureTreshold = 17;
    highPressureTreshold = 21;
    alarmOn = false;
}

void Alarm::check()
{
    double psiPressureValue = sensor.popNextPressurePsiValue();

    if (psiPressureValue < lowPressureTreshold || highPressureTreshold < psiPressureValue)
    {
        alarmOn = true;
    }
}

bool Alarm::isAlarmOn()
{
    return alarmOn;
}

File: php/src/TirePressureMonitoring/Alarm.php
<?php

declare(strict_types=1);

namespace RacingCar\TirePressureMonitoring;

class Alarm
{
    private const LOW_PRESSURE_THRESHOLD = 17;

    private const HIGH_PRESSURE_THRESHOLD = 21;

    private $sensor;

    private $isAlarmOn = false;

    public function __construct()
    {
        $this->sensor = new Sensor();
    }

    public function check(): void
    {
        $psiPressureValue = $this->sensor->popNextPressurePsiValue();

        if ($psiPressureValue < self::LOW_PRESSURE_THRESHOLD || $psiPressureValue > self::HIGH_PRESSURE_THRESHOLD) {
            $this->isAlarmOn = true;
        }
    }

    public function isAlarmOn(): bool
    {
        return $this->isAlarmOn;
    }
}
